src/emailpro/external-contact/add/emailpro-external-contact-add.controller.js

```diff
@@ -0,0 +1,110 @@
+angular
+    .module("Module.emailpro.controllers")
+    .controller("EmailProAddExternalContactCtrl", class EmailProAddExternalContactCtrl {
+        constructor ($scope, $stateParams, EmailPro, EmailProExternalContacts, Alerter, translator) {
+            this.$scope = $scope;
+            this.$stateParams = $stateParams;
+            this.EmailPro = EmailPro;
+            this.EmailProExternalContacts = EmailProExternalContacts;
+            this.Alerter = Alerter;
+            this.translator = translator;
+        }
+
+        $onInit () {
+            this.model = {
+                hiddenFromGAL: false
+            };
+
+            this.isLoading = false;
+            this.emailValidationRegex = "^[a-zA-Z0-9]+([\.+\-\w][a-zA-Z0-9]+)*@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$";
+
+            this.EmailPro
+                .getSelected()
+                .then((exchange) => {
+                    if (exchange.serverDiagnostic.version === 14 && exchange.offer === this.EmailPro.accountTypeProvider) {
+                        this.EmailProExternalContacts.getContactOptions(this.$stateParams.organization, this.$stateParams.productId)
```

Comment:
You should chain your promises and avoid using Promise without return it.

```
this.emailPro.getSlected()
                     .then(exchange =>{
                          // return your promise
                      })
                     .then(data => {
                        ...
                     })
                     .catch(failure => {
                        ...
                     })
```

The difference is : if your request `Email.getSelected()` fails, it will be catched ;)